lms/tasks/celery.py

```diff
@@ -38,6 +38,35 @@
         "interval_start": 0.2,
         "interval_step": 0.2,
     },
+    # Tell Celery to kill any task run (by raising
+    # celery.exceptions.SoftTimeLimitExceeded) if it takes longer than
+    # task_soft_time_limit seconds.
+    #
+    # See: https://docs.celeryq.dev/en/stable/userguide/workers.html#time-limits
+    #
+    # This is to protect against task runs hanging forever which blocks a
+    # Celery worker and prevents Celery retries from kicking in.
+    #
+    # This can be overridden on a per-task basis by adding soft_time_limit=n to
+    # the task's @app.task() arguments.
+    #
+    # We're using soft rather than hard time limits because hard time limits
+    # don't trigger Celery retries whereas soft ones do. Soft time limits also
+    # give the task a chance to catch SoftTimeLimitExceeded and do some cleanup
+    # before exiting.
+    task_soft_time_limit=120,
```

Comment:
Looking at how long LMS's Celery tasks take to run in New Relic I think two minutes should be loads of time:

| Task | 99% time for the last 3 months |
| ---- | ------------------------------ |
| `send_instructor_email_digests_tasks()` | 3.8s |
| `send_instructor_email_digests()` | 10.6s |
| `send_template()` | 613ms |
| `rotate_keys()` | 307ms |
| `delete_expired_rows()` | 59.3ms |

I took our longest running task (10.6s), multiplied it by 10, and rounded that up to 120s.